### DISCUSSION TITLE
Credit Cards

### DISCUSSION THREAD

**Discussion by albaloo2000** [firefly-iii/firefly-iii#6334] [General]
Hello all, 

I am brand new to firefly and just finished installing it.  Very impressive amount of effort has gone into this software and I was really happy to see its based on laravel as I am pretty familiar with it. 

However as I am trying to get started I am completely thrown off by the whole credit card as an asset and management of it.  A lot of personal finances these days involve credit cards.  I saw a post where someone mentions setting up the cards as assets and treat them as overdraft accounts but I can't figure out how to do that either. 

I searched on this discussions board as well but can't find anything.  Is there a step by step guide to show you how to set this up and how to deal with credit cards? 

@JC5 I haven't spent a lot of time in Firefly but it is so clear how much work and effort you have put into this software.  What are your thoughts on the credit cards and is there anyway to add dedicated support for credit cards in the software ?

**Comment by JC5** [comment]
Thanks for the compliments!

Firefly III is pretty opinionated on this. Every accounting tool considers credit cards a liability because you borrow the money. I consider that a boring technicality. To me, a credit card is an "asset account" just like my usual debit cards (take note: I'm not from the US).

So if I spend money on my CC I can either see it drop into a negative amount, or I set the virtual balance to whatever my credit limit is. Every month, I transfer the money spent from my normal checking account to the credit card, bringing it back to "zero".

That's basically the idea. I'm fairly sure it's in de documentation somewhere 😅

**Comment by albaloo2000** [comment]
That makes sense.  Does that mean you are entering each purchase as a negative item?   In that case if you do a CSV import you have to go through and change each number so its a negative?

**Reply by JC5** [reply]
No that's not necessary. The amount can be positive in your CSV file. The source account and the destination is what's important.

Is the amount always positive in your CSV files, also when it's a deposit?

**Comment by albaloo2000** [comment]
Generally credit cards show the purchases and the balances as a positive number.   So as you spend money on a card your balance goes up.  Payments to a credit card show up as negative numbers that bring that balance back down.  This is why most accounting software treats them as liabilities so the higher the balance, the more you owe.   CSV files from credit card companies also work the same way. Purchases are shown as positive and payments as negative.

I think your method would required going through each transaction and negating it.

**Reply by marcelbrueckner** [reply]
> Generally credit cards show the purchases and the balances as a positive number.

Each German bank I have or had a credit card show purchases with a credit card as negative as they do with regular asset accounts.

**Reply by JC5** [reply]
> I think your method would required going through each transaction and negating it.

I don't think that would be necessary. If you can share some (censored) lines from a CSV file, I can check out if the importer can handle them without editing them manually.

**Reply by albaloo2000** [reply]
That is starting to make a lot more sense for how you wrote the software.  This must be a difference between German banks and the US banks.  It also makes sense if you look at all the comments from different discussions about the confusion from the US crowd.   I saw one comment which literally said the more money I spend it looks like my assets are going up in FireFly

Yes in the US every single credit card I have ever had treats the numbers this way.  The good news though is I think it should be a fairly easy setting to do to automatically negate all those numbers for the people here in the us right?

**Reply by albaloo2000** [reply]
Here is a sample of a CVS file I get,  I changed the names but they show up as either me or my wife's name. 

Status,Date,Description,Debit,Credit,Member Name
Cleared,07/10/2022,"CNK*CINEMARK.COM 260 800-246-3627 FL",72.77,,FIRSTNAME LASTNAME
Cleared,07/09/2022,"SUNDAY* VIANA BARCELONA ES",82.67,,FIRSTNAME LASTNAME
Cleared,07/09/2022,"WPY*ORRA Staff 855-999-3729 TN",105.50,,FIRSTNAME LASTNAME
Cleared,07/08/2022,"CASA BATLLO S.L. BARCELONA ES",71.56,,FIRSTNAME LASTNAME
Cleared,07/08/2022,"Tiqets.com Amsterdam NL",51.12,,FIRSTNAME LASTNAME
Cleared,07/06/2022,"AUTOPAY 999990000009422RAUTOPAY AUTO-PMT",,-1000.94,FIRSTNAME1 LASTNAME1
Cleared,07/06/2022,"VIDAL PONS S A BARCELONA ES",8.47,,FIRSTNAME LASTNAME
Cleared,07/06/2022,"SAGRADA FAMILIA WEB B2C BARCELONA ES",56.87,,FIRSTNAME LASTNAME
Cleared,07/06/2022,"www.headout.com London DE",48.07,,FIRSTNAME LASTNAME

**Reply by JC5** [reply]
Of course, that setting is present in the importer. I think for 6 or 7 years now? This file should present you with no troubles when importing.

**Reply by albaloo2000** [reply]
It worked! Thank you for your help.  I did have to import using amount (negated).   Seems trivial but I don't think I would have figured this out on my own.  Thanks again for your help.

**Reply by JC5** [reply]
Glad to hear it, no problem!

**Comment by gbroin** [comment]
Hello,

impressive project, thank you very much for the contribution!

I wanted to know how to record credit card installment payments. Thanks!
